File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Model/Cita.java
package com.example.BackendCitasMedicas.Model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Cita {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id_cita;

    @Column(name = "fecha_hora", nullable = false)
    private LocalDateTime fecha_hora;
    private String estado;

    @ManyToOne
    @JoinColumn(name = "id_paciente")
    private Paciente paciente;

    @ManyToOne
    @JoinColumn(name = "id_medico")
    private Medico medico;

}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Model/Cita.kt
package com.example.frontcitasmedicas.Model

import java.time.LocalDateTime

data class Cita(
    val id_cita: Long = 0,
    val fecha_hora: String = "", // Se puede manejar como String o LocalDateTime, depende de cómo lo recibas en la API
    val estado: String = "",
    val paciente: Paciente? = null,
    val medico: Medico? = null
)